packages/web-components/fast-foundation/src/picker/picker.ts

```diff
@@ -95,22 +95,62 @@ export class FASTPicker extends FormAssociatedPicker {
     }
 
     /**
-     * Whether the component should remove an option from the list when it is in the selection
+     * Deprecated. Whether the component should remove an option from the list when it is in the selection. Default is false.
      *
-     * @remarks
-     * HTML Attribute: filter-selected
+     * @deprecated - use `Picker.disableSelectionFilter`.
      */
-    @attr({ attribute: "filter-selected", mode: "boolean" })
-    public filterSelected: boolean = true;
+    @observable
+    public filterSelected: boolean = false;
+    protected filterSelectedChanged(): void {
+        // keep the main prop in sync
+        if (this.disableSelectionFilter === this.filterSelected) {
+            this.disableSelectionFilter = !this.filterSelected;
+        }
+    }
 
     /**
-     * Whether the component should remove options based on the current query
+     * Whether the component should remove an option from the list when it is in the selection. Default is false.
      *
      * @remarks
-     * HTML Attribute: filter-query
+     * HTML Attribute: disable-selection-filter
      */
-    @attr({ attribute: "filter-query", mode: "boolean" })
+    @attr({ attribute: "disable-selection-filter", mode: "boolean" })
+    public disableSelectionFilter: boolean = false;
+    protected disableSelectionFilterChanged(): void {
+        // keep depracated prop in sync
+        if (this.disableSelectionFilter === this.filterQuery) {
+            this.filterSelected = !this.disableSelectionFilter;
+        }
+    }
```

Comment:
Is it correct that it's comparing `this.disableSelectionFilter` to `this.filterQuery` here? If not, this can be simplified to be more readable:

```suggestion
    protected disableSelectionFilterChanged(prev: boolean | undefined, next: boolean): void {
        // keep deprecated prop in sync
        if (this.filterSelected === next) {
            this.filterSelected = !next;
        }
    }
```

Reply:
Personally, I find using the prop name more readable and succinct than using "next".

Reply:
The issue in this particular block is two parts.

The first is the switch to `prev`/`next`, which is primarily a stylistic decision but it also reduces our own use of deprecated properties. Additionally, I had to write out an algorithm summary for each method to understand what they were doing, since there are so many properties with very similar names and I got confused while reading the changes.

As a result of writing out the algorithm summaries, I found this specific second issue where the assertion on line 121 seems to be incorrect, but I might be wrong:
```diff
-        if (this.disableSelectionFilter === this.filterQuery) {
+        if (this.disableSelectionFilter === this.filterSelected) {
```

Reply:
How does the other mirror prop change otherwise?  I'm wondering if we should just make this a breaking change and remove the legacy prop/attributes since we're doing a major version bump.

As for the readability question, I am not at all convinced that 

`protected myPropChanged(prev, next): void {
    if (next === undefined) {
      ... do something
   }
}`

should be preferred over 

`protected myPropChanged(): void {
    if (this.myProp === undefined) {
      ... do something
   }
}`
 
I prefer the second one and generally only use the params if I care about what "prev" was in my change handler.

Reply:
This seems to be preference, closing it out as part of #6951